core/modules/util/ConfigStore.cc

```diff
@@ -0,0 +1,142 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2014-2016 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "ConfigStore.h"
+
+// System headers
+#include <map>
+#include <sstream>
+
+// Qserv headers
+#include "util/ConfigStoreError.h"
+#include "util/IterableFormatter.h"
+
+// Third-party headers
+#include "boost/lexical_cast.hpp"
+#include "boost/property_tree/ini_parser.hpp"
+#include "boost/property_tree/ptree.hpp"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+
+namespace { // File-scope helpers
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.util.ConfigStore");
+
+} // namespace
+
+
+namespace lsst {
+namespace qserv {
+namespace util {
+
+
+std::map<std::string, std::string> const ConfigStore::_parseIniFile(std::string const& configFilePath) {
+
+    // read it into a ptree
+    boost::property_tree::ptree pt;
+    boost::property_tree::ini_parser::read_ini(configFilePath, pt);
+
+    std::map<std::string, std::string> configMap;
+
+    // flatten
+    for (auto& sectionPair: pt) {
+        auto& section = sectionPair.first;
+        for (auto& itemPair: sectionPair.second) {
+            auto& item = itemPair.first;
+            auto& value = itemPair.second.data();
+            configMap.insert(std::make_pair(section + "." + item, value));
+        }
+    }
+
+    return configMap;
+}
+
+std::string ConfigStore::getRequired(std::string const& key) const {
+    std::map<std::string, std::string>::const_iterator i = _configMap.find(key);
```

Comment:
Fabrice, I already mentioned this on other comment - {{_configMap}} is const (because method is const) so find() will return const_iterator. Try to replace const_iterator with iterator - it will not compile.